testutil/keeper/tokenomics.go

```diff
@@ -117,21 +144,174 @@ func TokenomicsKeeper(t testing.TB) (
 	mockAccountKeeper := mocks.NewMockAccountKeeper(ctrl)
 	mockAccountKeeper.EXPECT().GetAccount(gomock.Any(), gomock.Any()).AnyTimes()
 
-	tokenomicsKeeper := keeper.NewKeeper(
+	// Mock the proof keeper
+	mockProofKeeper := mocks.NewMockProofKeeper(ctrl)
+	mockProofKeeper.EXPECT().GetAllClaims(gomock.Any()).AnyTimes()
+
+	k := tokenomicskeeper.NewKeeper(
 		cdc,
 		runtime.NewKVStoreService(storeKey),
 		log.NewNopLogger(),
 		authority.String(),
 		mockBankKeeper,
 		mockAccountKeeper,
 		mockApplicationKeeper,
-		mockSupplierKeeper,
+		mockProofKeeper,
 	)
 
 	ctx := sdk.NewContext(stateStore, cmtproto.Header{}, false, log.NewNopLogger())
 
 	// Initialize params
-	require.NoError(t, tokenomicsKeeper.SetParams(ctx, types.DefaultParams()))
+	require.NoError(t, k.SetParams(ctx, tokenomicstypes.DefaultParams()))
+
+	return k, ctx, application.Address, supplier.Address
+}
+
+// NewTokenomicsModuleKeepers is a helper function to create a tokenomics keeper and a context. It uses
+// real dependencies for all keepers except the bank keeper, which is mocked as it's not used
+// directly by the tokenomics keeper or its dependencies.
+func NewTokenomicsModuleKeepers(t testing.TB, opts ...TokenomicsKeepersOpt) (_ TokenomicsModuleKeepers, ctx context.Context) {
```

Comment:
AFAIK, the reason for making `ctx` a named variable here is so that `sdk.NewContext` can override it as the `context.Context` interface type, as opposed to `sdk.Context` concrete type, and to avoid extra variable declarations or casting.

Reply:
Keeping as is.